ryuseisan/ssl-mlflow
# Add nginx reverse proxy with SSL for mlflow (#1)

### Description
This PR introduces an nginx reverse proxy setup to provide SSL support for our mlflow tracking server. With this setup, the nginx service listens on port 443 and proxies requests to the mlflow service, which runs on port 5000.

### Changes
- **compose.yaml**: Added an nginx service. This service uses the latest nginx image and has SSL configuration. SSL certificates are mounted from the local `./ssl` directory.
- **Dockerfile**: Updated the base image to the latest Python and specified the mlflow version to be 2.6.0 or later.
- **nginx.conf**: Introduced this configuration to set up SSL for the nginx server. It also contains settings for proxying requests to the mlflow service.

### Testing
1. Build and run the docker-compose setup: `docker compose up -d`
2. Access the mlflow server via HTTPS using a browser or a tool like curl: `https://localhost`
3. Confirm that the connection is secured with SSL.

### Additional Notes
Ensure that the SSL certificates are correctly placed in the `./ssl` directory before starting the services. Future improvements might include automating the certificate renewal process and further refining the nginx settings for better performance and security.
